File: frontend/src/services/taskService.js
import axios from 'axios'

export const getTasks = async (params = {}) => (await axios.get('/api/tasks', { params })).data
export const getTasksForProject = async (projectId) => (await axios.get(`/api/tasks/project/${projectId}`)).data
export const createTask = async (payload) => (await axios.post('/api/tasks', payload)).data
export const updateTask = async (taskId, payload) => (await axios.put(`/api/tasks/${taskId}`, payload)).data
export const deleteTask = async (taskId) => axios.delete(`/api/tasks/${taskId}`)

File: backend/src/main/java/com/rashid/backend/service/interfaces/TaskService.java
package com.rashid.backend.service.interfaces;

import com.rashid.backend.dto.common.PagedResponseDTO;
import com.rashid.backend.dto.task.TaskDTO;
import com.rashid.backend.dto.task.TaskFilterDTO;

public interface TaskService {
    TaskDTO createTask(TaskDTO taskDTO, String username);
    PagedResponseDTO<TaskDTO> getTasksForProject(Long projectId, int page, int size, String username);
    PagedResponseDTO<TaskDTO> getTasks(TaskFilterDTO filter, int page, int size, String username);
    TaskDTO updateTask(Long taskId, TaskDTO taskDTO, String username);
    void deleteTask(Long taskId, String username);
}
